### PR TITLE
Add recurring task support

### PR DESCRIPTION
**Pull request by mitsuhiko**
This adds support for running functions in a background thread in a recurring
manner.  Functions can be defined to run for certain services and will be
executed on each machine.  This is used for supporting things like dsym
cache clearing that are local to the machine.

### CODE REVIEW

**Comment by mattrobenolt** [comment]
How will this work when there are multiple worker processes running on a single machine? Wouldn't we only want one per physical host?

File: src/sentry/utils/recurring.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            )
+            if config['interval'] is None:
+                raise RuntimeError('No interval set for %s' % task_id)
+            tasks.setdefault(service, {})[task_id] = config
+
+
+def run_recurring_for_service(service):
+    logger.info('Starting recurring tasks for %s', service)
+    load_config()
+    to_run = tasks.setdefault(service, {})
```
Comment (mattrobenolt) [inline src/sentry/utils/recurring.py]: Seems we should be able to return early here if there are no tasks to run for this service, right? And avoid spawning a thread always with nothing to do.

**Comment by mitsuhiko** [comment]
You are right that ideally one can tell this to only run once per machine. Right now that is not a super big concern for dsyms. That's also why I added the jitter.

**Comment by markstory** [comment]
Is this something we're still interested in pursuing or can it be closed because it can no longer be merged?

**Pull request closed without merging**